config/default.go

```diff
@@ -351,3 +298,83 @@ func DefaultFifoFile() (string, error) {
 	}
 	return filepath.Join(f, "gpud.fifo"), nil
 }
+
+func DefaultContainerdComponent(ctx context.Context) (any, bool) {
+	if runtime.GOOS != "linux" {
+		log.Logger.Debugw("ignoring default containerd pod checking since it's not linux", "os", runtime.GOOS)
+		return nil, false
+	}
+
+	containerdSocketExists := false
+	containerdRunning := false
```

Comment:
for containerd/docker when its not running, I think component also need to be registered. just unhealthy, and for kubelet, we can use pidof kubelet instead of check 10255, the rest lgtm

Reply:
done